src/main/java/org/folio/services/finance/transaction/PendingPaymentWorkflowService.java

```diff
@@ -137,11 +137,14 @@ private List<InvoiceWorkflowDataHolder> withNewPendingPayments(List<InvoiceWorkf
   }
 
   private Future<Void> updateTransactions(List<InvoiceWorkflowDataHolder> holders, RequestContext requestContext) {
-    var futures = holders.stream()
-      .map(InvoiceWorkflowDataHolder::getNewTransaction)
-      .map(transaction -> baseTransactionService.updateTransaction(transaction, requestContext))
-      .collect(Collectors.toList());
-    return GenericCompositeFuture.join(futures).mapEmpty();
```

Comment:
So the issue was in parallel processing of transactions?

Reply:
I am not exactly sure why primary key violation when we use java stream . it may be because of parallel processing. when i try to update transactions one by one using for each ,it is running fine .